JumpscaleLibs/sal/zosv2/billing.py

```diff
@@ -0,0 +1,60 @@
+from .resource import ResourceParser
+
+
+class Billing:
+    def __init__(self, explorer):
+        self._explorer = explorer
+
+    def reservation_resources(self, reservation):
+        """
+        compute how much resource units is reserved in the reservation
+
+        :param reservation: reservation object
+        :type reservation: tfgrid.workloads.reservation.1
+        :return: list of ResourceUnitsNode object
+        :rtype: list
+        """
+        rp = ResourceParser(self._explorer, reservation)
+        return rp.calculate_used_resources()
+
+    def reservation_resources_cost(self, reservation):
+        """
+        compute how much resource units is reserved in the reservation
+
+        :param reservation: reservation object
+        :type reservation: tfgrid.workloads.reservation.1
+        :return: list of ResourceUnitsNode object with costs filled in
+        :rtype: list
+        """
+        rp = ResourceParser(self._explorer, reservation)
+        return rp.calculate_used_resources_cost()
+
+    def payout_farmers(self, tfchain_wallet, resource_units_per_node, reservation):
+        """
+        payout farmer based on the resources per node used
+
+        :param tfchain_wallet: tfchain wallet
+        :type tfchain_wallet: a wallet of j.clients.tfchain
+        :param resource_units_per_node: list of resource units per node retrieved from reservation_resources_cost
+        :type resource_units_per_node: list of ResourceUnitsNode
+        :param reservation: reservation object
+        :type int
+        :return: list of transactions
+        :rtype: list
+        """
+        rp = ResourceParser(self._explorer, reservation)
+        return rp.payout_farmers(tfchain_wallet, resource_units_per_node, reservation.id)
+
+    def verify_payments(self, tfchain_wallet, reservation):
```

Comment:
Not sure if we should ask the farmer to provide an entire reservation object here, or if we should just accept the reservation ID and fetch it ourselves

Reply:
I want both side of the payement flow to look the same.
So I added a method to get a reservation from ID
so both `verify_payments` and `payout_farmers` have the same signature